ITR/data/base_providers.py

```diff
@@ -107,21 +126,38 @@ def __init__(self, EI_benchmarks: IEIBenchmarkScopes,
         self._EI_benchmarks = EI_benchmarks
         self.temp_config = tempscore_config
         self.column_config = column_config
+        benchmarks_as_series = []
+        for scope in EScope.get_result_scopes():
+            try:
+                for bm in getattr(EI_benchmarks, scope.name).benchmarks:
+                    benchmarks_as_series.append(self._convert_benchmark_to_series(bm, scope))
+            except AttributeError:
+                pass
```

Comment:
if it's an `AttributeError` - maybe we shouldn't `pass`.
maybe we better `raise` an exception as usual

Reply:
So I've been reading a lot about Python's "better to ask forgiveness than permission" (which is opposite to C++).  In this case, we are using `getattr` which can raise AttributeError if the attribute we are looking for does not exist.  In the case of TPI, which typically only has a benchmark for a single scope definition (S1, S1S2, S3, or S1S2S3) the getattr will fail on all scope queries except one.  I could have written the code to ask for permission, but instead wrote it to ask for forgiveness.  If there is no attribute, there is no error, we just move along...